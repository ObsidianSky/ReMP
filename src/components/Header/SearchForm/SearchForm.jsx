import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './SearchForm.scss'
import Input from '../../common/Input/Input';
import Button from '../../common/Button/Button';
import RadioButtonGroup from '../../common/RadioButtonGroup/RadioButtonGroup';

import { searchMovies, setSearchQuery, setSearchType } from '../../../actions';

export class SearchForm extends Component {
    constructor(props) {
        super(props);
        this.onFormSubmit = this.onFormSubmit.bind(this);
    }

    onFormSubmit(e) {
        e.preventDefault();
        this.props.onSearch();
    }

    render() {
        return (
            <form className="search-form" onSubmit={this.onFormSubmit}>
                <div className="search-form__title">Find your movie</div>
                <div className="search-form__error">{this.props.error}</div>
                <div className="search-form__input">
                    <Input value={this.props.query} onChange={this.props.onSearchQueryChange}/>
                </div>
                <div className="search-form__controls">
                    <RadioButtonGroup
                        activeOption={this.props.type}
                        options={this.props.searchTypes}
                        onSelect={this.props.onSearchTypeChange}
                        title="Search by"
                    />
                    <Button className="btn--wide btn--pink" onClick={this.props.onSearch}>Search</Button>
                </div>
            </form>
        )
    }
}

SearchForm.propTypes = {
    onSearchQueryChange: PropTypes.func.isRequired,
    onSearchTypeChange: PropTypes.func.isRequired,
    onSearch: PropTypes.func.isRequired,
    query: PropTypes.string,
    error: PropTypes.string,
    type: PropTypes.string,
    searchTypes: PropTypes.arrayOf(PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
    })).isRequired
};

const mapStateToProps = ({ search }) => ({
    type: search.type,
    searchTypes: search.types,
    query: search.query,
    error: search.error
});

const mapDispatchToProps = {
    onSearchQueryChange: setSearchQuery,
    onSearchTypeChange: setSearchType,
    onSearch: searchMovies
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);
