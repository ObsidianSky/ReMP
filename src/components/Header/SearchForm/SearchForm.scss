@import '../../../assets/styles/variables';

.search-form {
  &__title {
    margin-bottom: 20px;
    text-transform: uppercase;
    color: $text-light;
    font-size: 22px;
  }

  &__input {
    margin-bottom: 15px;
  }

  &__controls {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
}