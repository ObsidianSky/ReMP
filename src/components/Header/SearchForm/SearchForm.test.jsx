import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { SearchForm } from './SearchForm';

describe('SearchForm', () => {
    let props;
    let onSearch;

    beforeEach(() => {
        onSearch = jest.fn();

        props = {
            onSearchQueryChange: () => {},
            onSearchTypeChange: () => {},
            onSearch,
            query: 'query',
            error: 'error',
            type: 'type',
            searchTypes: [
                { label: 'title', value: 'title' },
                { label: 'director', value: 'director' }
            ]
        }
    });

    test('should renders correctly', () => {
        const wrapper = shallow(<SearchForm { ...props }/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('on form submit', () => {
        let preventDefault;
        let wrapper;

        beforeEach(() => {
            preventDefault = jest.fn();
            wrapper = shallow(<SearchForm { ...props }/>);
        });

        test('should prevent default form behaviour', () => {
            wrapper.find('.search-form').simulate('submit', { preventDefault });

            expect(preventDefault).toHaveBeenCalled();
        });

        test('should call on search callback', () => {
            wrapper.find('.search-form').simulate('submit', { preventDefault });

            expect(onSearch).toHaveBeenCalled();
        });
    });
});