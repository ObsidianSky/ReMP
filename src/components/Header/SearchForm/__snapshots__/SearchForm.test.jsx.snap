// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should renders correctly 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <div
    className="search-form__title"
  >
    Find your movie
  </div>
  <div
    className="search-form__error"
  >
    error
  </div>
  <div
    className="search-form__input"
  >
    <Input
      onChange={[Function]}
      placeholder=""
      value="query"
    />
  </div>
  <div
    className="search-form__controls"
  >
    <RadioButtonGroup
      activeOption="type"
      onSelect={[Function]}
      options={
        Array [
          Object {
            "label": "title",
            "value": "title",
          },
          Object {
            "label": "director",
            "value": "director",
          },
        ]
      }
      title="Search by"
      type="search"
    />
    <Button
      className="btn--wide btn--pink"
      onClick={[Function]}
      text=""
    >
      Search
    </Button>
  </div>
</form>
`;
